e2e/app/aou-elements/WebElement.ts

```diff
@@ -0,0 +1,175 @@
+import {ClickOptions, ElementHandle, Page, WaitForSelectorOptions} from 'puppeteer';
+import {ElementInterface} from './ElementInterface';
+
+export default class WebElement implements ElementInterface {
+
+  public static asWebElement(page: Page, elem: ElementHandle): WebElement {
```

Comment:
After reviewing more deeply, I see the potential use for the xpath utilities. However, I'm not convinced about the "WebElement / AouElement". Can you summarize in the PR description, or a doc, or a README the high level design of this framework?

Here's I'm seeing, and here's my current thinking:
- We have a "Page" concept, which wraps standard functionality / higher level page interactions. I buy the need for this.
- We have a low level xpath utility library for easier xpath creation. I think I buy the need for this *if* we're using xpath prolifically (using data-test-id more heavily reduces the need). xpath is maybe still fine for some things like picking out display strings.
- Lastly we have this WebElement / AoUElement. From what I can tell, this is effectively wrapping functionality found in ElementHandle or Page. Why are we not just directly using ElementHandle, which is the interface provided by Puppeteer? If there is common functionality we want to reuse, it may be sufficient to just use static utility helpers, rather creating an intermediate wrapper over the Puppeteer model. The advantage of using a standard API is that is a fair amount of existing documentation / support material for using it.

There are a lot of lines of code here, and a lot of things to maintain / keep in sync. I also don't want to add unnecessary wrapping / boilerplate when we can use standard interfaces. The cost might be justified in certain cases, but we want to document when/why we're making those decisions. I think a high level README.md would help here. If you're basing this off a pattern in another web testing framework, please make that clear.

Reply:
explanation and reasons in JIRA ticket hopefully answer some of your questions here. The main reason for wrapper `WebElement` is I want to use the dot `.` notation to access useful functions instead import static functions.